packages/auth/src/auth-repo.ts

```diff
@@ -3,9 +3,11 @@ import type { User } from './user';
 export type AuthRepoBuildLoginUrlParams = { returnTo?: string };
 
 export interface AuthRepo {
-  buildLoginUrl(params?: AuthRepoBuildLoginUrlParams): string;
-
   isAuthenticated(): Promise<boolean>;
 
   getUser(): Promise<User>;
+
+  requestLogin(): void;
+
+  onUserAuthenticated(callback: (user: User) => void): void;
```

Comment:
- Reuse types
- Add Unsubscription behavior

```suggestion
  onUserAuthenticated(callback: UserAuthenticatedCallback): UnsubscribeCallback;
```

```ts
type UserAuthenticatedCallback = (user: User) => void;
type UnsubscribeCallback = () => void;
```

Reply:
Not sure I follow your idea with `UnsubscibeCallback`. Do you expect something like this?

```
onUserAuthenticated(callback: UserAuthenticatedCallback): UnsubscribeCallback {
    this.onUserAuthenticatedCallback = callback;
    return () => {this.onUserAuthenticatedCallback = ()=> {}}
  }
```

Return a function to clear the callback?